### PR TITLE
[NG] fix(datagrid-filter): Removes filter icon svg from scss

### PR DESCRIPTION
**Pull request by hippee-lee**
This adds clr-icon to the filter template and removes the filter svg from the scss. It also fixes issue #3341 that was causing an error in IE/Edge when filters were active.

Signed-off-by: Matt Hippely <mhippely@vmware.com>

### CODE REVIEW

File: src/clr-angular/image/_icons.clarity.scss [hunk 1]
```diff
@@ -76,13 +76,6 @@ $svg_data: 'data:image/svg+xml;charset=utf8';
...
 }
 
-@function generateFilterIcon($fillColor: $clr-color-action-600) {
-  $filter-off-icon: '%3Csvg%20xmlns%3D%22http%3A%2F%2Fwww.w3.org%2F2000%2Fsvg%22%20viewBox%3D%220%200%2016%2016%22%3E%3Cdefs%3E%3Cstyle%3E.cls-1%7Bfill%3A%23' +
-    str-slice($fillColor + '', 2) +
-    '%3Bfill-rule%3Aevenodd%3B%7D%3C%2Fstyle%3E%3C%2Fdefs%3E%3Cpolygon%20class%3D%22cls-1%22%20points%3D%227%209.32%203%205.38%203%205%2013%205%2013%205.38%209%209.32%209%2012.21%207%2013.29%207%209.32%22%2F%3E%3C%2Fsvg%3E';
-  @return url('#{$svg_data},#{$filter-off-icon}');
-}
-
 @function generateCircleIcon($fillColor: $clr-color-action-600) {
```
Comment (mathisscott) [inline src/clr-angular/image/_icons.clarity.scss]: I believe you have also removed the only usage of `generateCircleIcon` in our library and can safely remove this function as well...

Comment (hippee-lee) [reply src/clr-angular/image/_icons.clarity.scss]: Right, I can't find any usages for it. I'll remove and push a second commit.

Comment (mathisscott) [reply src/clr-angular/image/_icons.clarity.scss]: 😂 That was fast!

Great work. I'm happy to see these start going away.

File: src/clr-angular/data/datagrid/datagrid-filter.ts [hunk 2]
```diff
@@ -25,8 +25,9 @@ import { ClrCommonStrings } from '../../utils/i18n/common-strings.interface';
   providers: [{ provide: CustomFilter, useExisting: ClrDatagridFilter }],
   template: `
         <button #anchor class="datagrid-filter-toggle" (click)="toggle()"
-            [class.datagrid-filter-open]="open" [class.datagrid-filtered]="active"
-            type="button"></button>
+                [class.datagrid-filter-open]="open" [class.datagrid-filtered]="active">
+            <clr-icon [attr.shape]="active ? 'filter-grid-circle': 'filter-grid'" class="is-solid"></clr-icon>
+        </button>
```
Comment (coryrylan) [inline src/clr-angular/data/datagrid/datagrid-filter.ts]: Was the `type="button"` supposed to be removed?

Comment (hippee-lee) [reply src/clr-angular/data/datagrid/datagrid-filter.ts]: nope. I forgot to put it back once i got it working. Will add it.

**Pull request merged**